gufe/protocols/protocol_settings.py

```diff
@@ -0,0 +1,381 @@
+        arbitrary_types_allowed = True
+
+    sampler_method = "repex"
+    online_analysis_interval: Optional[int] = None
+    online_analysis_target_error = 0.2 * unit.boltzmann_constant * unit.kelvin
+    online_analysis_minimum_iterations = 50
+    flatness_criteria = 'logZ-flatness'
+    gamma0 = 0.0
+
+    @validator('online_analysis_target_error',
+               'online_analysis_minimum_iterations', 'gamma0')
+    def must_be_positive(cls, v):
+        if v < 0:
+            errmsg = ("Online analysis target error, minimum iteration "
+                      "and SAMS gamm0 must be 0 or positive values")
+            raise ValueError(errmsg)
+        return v
+
+
+class BarostatSettings(BaseModel):
+    """Settings for the OpenMM Monte Carlo barostat series
+
+    Attributes
+    ----------
+    pressure : float * unit.bar
+      Target pressure acting on the system. Default 1 * unit.bar.
+    frequency : int * unit.timestep
+      Frequency at which volume scaling changes should be attempted.
+      Default 25 * unit.timestep.
+
+    Notes
+    -----
+    * The temperature is defined under IntegratorSettings
+
+    TODO
+    ----
+    * Add support for anisotropic and membrane barostats.
+    """
+    class Config:
+        arbitrary_types_allowed = True
+
+    pressure = 1 * unit.bar
+    frequency = 25 * unit.timestep
+
+    @validator('pressure')
+    def must_be_positive(cls, v):
+        if v <= 0:
+            raise ValueError("Pressure must be positive")
+        return v
+
+    @validator('pressure')
+    def is_pressure(cls, v):
+        if not v.is_compatible_with(unit.bar):
+            raise ValueError("Must be pressure value, e.g. use unit.bar")
+        return v
+
+
+class IntegratorSettings(BaseModel):
+    """Settings for the LangevinSplittingDynamicsMove integrator
+
+    Attributes
+    ----------
+    timestep : float * unit.femtosecond
+      Size of the simulation timestep. Default 2 * unit.femtosecond.
+    temperature : float * unit.kelvin
+      Target simulation temperature. Default 298.15 * unit.kelvin.
+    collision_rate : float / unit.picosecond
+      Collision frequency. Default 1 / unit.pisecond.
+    n_steps : int * unit.timestep
+      Number of integration timesteps each time the MCMC move is applied.
+      Default 1000.
+    reassign_velocities : bool
+      If True, velocities are reassigned from the Maxwell-Boltzmann
+      distribution at the beginning of move. Default False.
+    splitting : str
+      Sequence of "R", "V", "O" substeps to be carried out at each
+      timestep. Default "V R O R V".
+    n_restart_attempts : int
+      Number of attempts to restart from Context if there are NaNs in the
+      energies after integration. Default 20.
+    constraint_tolerance : float
+      Tolerance for the constraint solver. Default 1e-6.
+    """
+    class Config:
+        arbitrary_types_allowed = True
+
+    timestep = 2 * unit.femtosecond
+    temperature = 298.15 * unit.kelvin
```

Comment:
As with `BarostatSettings`, this convolutes thermodynamic parameters that impact the free energy of interest (`temperature`) with a mess of settings that impact the efficiency of sampling and stability of sampling. These should be separated, grouping temperature, pressure, etc. into the thermodynamic state information.

Reply:
I'd also highly recommend considering we adopt some flexible definition of how the sampling is done, perhaps in a hierarchical manner: It's most important to specify first (1) _what_ calculation do you want to perform (openff-2.0.0 at 300 K and 1 atm), then (2) how you computed it, at a high level (e.g. replica exchange, nonequilibrium switching), then (3) how exactly it is performed can be left to best practices defaults, auto-tuned, or specified explicitly if this specification can actually be implemented by the engine. But I imagine a great deal of work will be trying to automate best practices and auto-tuning for each calculation type.

More broadly, in openmmtools, for the lowest level (3), we have embraced a [general MCMC framework](https://openmmtools.readthedocs.io/en/latest/mcmc.html) that allows us to compose or chain together MCMC samplers that sample from the same alchemical state in order to efficiently and flexibly sample from within an alchemical state. This is a much more general design than specifying all the details of a single integrator, since a calculation may want to compose compatible types of sampling strategies.